gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -122,15 +132,55 @@ def self.signature_for_method(method)
     @signatures_by_method[key]
   end
 
+  # a module A such that:
+  #   if A is included or extended by a module B,
+  #   then we want B to have the hooks installed on itself
+  # should extend this module.
+  module VirallyInstallHooks
+    def included(arg)
+      super(arg)
+      if arg.is_a?(Module)
+        ::T::Private::Methods.install_hooks(arg)
+      end
+    end
```

Comment:
Can you think of some ways this could potentially break, and pre-empt them with test cases?

Like none of the test cases would break currently if when I remove `super`. Similarly none of them break if i remove the `is_a?` (it might be the case that that's safe to remove).

Reply:
I added some new tests (and in the process, discovered that whenever you call a final method, it thinks you're trying to redefine it).